packages/react-scripts/scripts/coalesceLocales.js

```diff
@@ -16,77 +16,107 @@ exports.coalesceLocales = paths => {
     rimraf.sync(builtLocalesDir);
   }
   fs.mkdirSync(builtLocalesDir);
-  
-  const index = fs.existsSync(`${paths.appSrc}/index.tsx`) ? `${paths.appSrc}/index.tsx` : `${paths.appSrc}/index.js`
-  
+
+  const index = fs.existsSync(`${paths.appSrc}/index.tsx`)
+    ? `${paths.appSrc}/index.tsx`
+    : `${paths.appSrc}/index.js`;
+
   const list = dependencyTree.toList({
     filename: index,
     directory: paths.appPath,
-    tsConfig: index.includes('.tsx') && `${paths.appPath}/tsconfig.json`,
+    tsConfig:
+      (index.includes('.tsx') || fs.existsSync(paths.appTsConfig)) &&
+      paths.appTsConfig,
```

Comment:
do we even need `index.includes('.tsx')`? It's probably a safe assumption that an index.tsx would have a tsconfig file so I guess it's harmless. But I don't see a scenario where we couldn't just check for the existence of the tsconfig file and apply it regardless. If you are trying to bypass checking for the tsconfig if this is a regular index.js file, this logic doesn't seem to do that to me. So with this logic, everything depends on if there is a tsconfig or not, the type of index doesn't really matter.
if we want the type of index to matter then we need to adjust the logic

Reply:
I left it in there to keep it possibly faster. I assume that the fs.existsSync is a bit slower. But I don't know how much of a difference that is so I would be fine with removing the index.includes part